### PR TITLE
Add ReceiveSendGreetingServlet and publish workflow for matrix project

### PR DESCRIPTION
**Pull request by samarth-gupta-traceable**
## Description
Add ReceiveSendGreetingServlet.java and publish workflow for matrix project

1. new servelt ReceiveSendGreetingServlet is capable of handling POST request data and echo is back
2. new git workflow for publishing the docker images for various servlet containers is triggered  only when changes are made to matrix project path and main branch. 

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

**Comment by pavolloffay** [comment]
Some files have OTEL license header and some not, could we add the license header to the missing files? We should preserve OTEL license header.

**Comment by samarth-gupta-traceable** [comment]
> Some files have OTEL license header and some not, could we add the license header to the missing files? We should preserve OTEL license header.

@pavolloffay  Below files do not have OTEL header (with reason for missing):

1. ExceptionRequestListener.java:   Source file in OTEL repo itself does not have the header. I can add it though if desirable. https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/smoke-tests/matrix/src/main/java/io/opentelemetry/smoketest/matrix/ExceptionRequestListener.java
2. GreetingWithHeader , StreamTransferUtil, ReceiveSendGreetingServlet : does not exists in OTEL repo and are added by us for HT. If needed I can add header in these as well.

**Comment by pavolloffay** [comment]
> https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/smoke-tests/matrix/src/main/java/io/opentelemetry/smoketest/matrix/ExceptionRequestListener.java

please add it to our copy. Also in the matrix readme mention that this is from OTEL

**Comment by samarth-gupta-traceable** [comment]
> > https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/smoke-tests/matrix/src/main/java/io/opentelemetry/smoketest/matrix/ExceptionRequestListener.java
> 
> please add it to our copy. Also in the matrix readme mention that this is from OTEL

done 👍

File: smoke-tests/matrix/src/main/java/io/opentelemetry/smoketest/matrix/ReceiveSendGreetingServlet.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import javax.servlet.http.HttpServletResponse;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.PrintWriter;
+import java.net.URL;
+import java.net.URLConnection;
+import java.util.stream.Collectors;
+
+public class ReceiveSendGreetingServlet extends HttpServlet {
```
Comment (pavolloffay) [inline smoke-tests/matrix/src/main/java/io/opentelemetry/smoketest/matrix/ReceiveSendGreetingServlet.java:19]: Could we rename it `EchoServlet`?

Comment (pavolloffay) [reply smoke-tests/matrix/src/main/java/io/opentelemetry/smoketest/matrix/ReceiveSendGreetingServlet.java:19]: Could do servlet just simply read the body to a buffer and write it back with the same content type? 

This way we don't need dep on Gson and additional classes. To test response heders capture the servlet could also send some test response header.

Comment (samarth-gupta-traceable) [reply smoke-tests/matrix/src/main/java/io/opentelemetry/smoketest/matrix/ReceiveSendGreetingServlet.java]: @pavolloffay  made below changes. 
1. renamed the servlet to EchoServlet
2. copying request as response . Header going as test headers.

**Review by pavolloffay — APPROVED** [review]

**Pull request merged**